### ISSUE TITLE
puppet agent fails to run ` Could not autoload puppet/type/wildfly_resource`

### ISSUE THREAD

**Issue by ashish1099** [biemond/biemond-wildfly#124]
What is happening sometime puppet fails to run and sometime it does not fail. I don't think the require statement under http_api file, would be creating any problem. any idea why this sudden burst of puppet

```
puppet agent t --environment=ashish
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Could not autoload puppet/type/wildfly_resource: Could not autoload puppet/provider/wildfly_resource/http_api: no such file to load - puppet_x/util/wildfly_cli on node beaker-role-appeng-wildfly-centos7-menywugg.ashish
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```

**Comment by jairojunior** [comment]
Sorry it took me so long to answer, got stuck with some personal issues.

Which version of Puppet Master are you using? I've been using it with a Puppet Master < 4 without problems.

**Comment by ashish1099** [comment]
@jairojunior hi sorry. missed the notification mail.

master and agent on puppet 3.8.x the issue occur randomly. like If I run puppet agent 4 times, it will fail 2 time or 3 time.

So not sure what could be the issue. I have tried to directly require the lib, but still. 

any idea :)

**Comment by aperto-armin** [comment]
Hi,

I also got the same error. I'm using it with Puppet Master 4.4.1.

**Comment by jairojunior** [comment]
@ashish1099 @aperto-armin Tried to reproduce this with both Puppet 3.8 and Puppet 4.4, but wasn't able.

Could you please provide more feedback? Ruby version maybe?

**Comment by ashish1099** [comment]
@jairojunior here it is 
In my case it is happening at random, so I when I'm testing with beaker it fails in the first run and sometime it doesn't. 

To me the code looks fine something else is wrong here. I sense

 ```[root@beaker-role-appeng-wildfly-centos7-otokdtwn ~]# ruby -v
ruby 2.0.0p598 (2014-11-13) [x86_64-linux]```

**Comment by aperto-armin** [comment]
@jairojunior my ruby version is 2.0.0p598

**Comment by nicolasbrechet** [comment]
I'm having the same behaviour with Puppet Enterprise 4.2.2.
Could it be linked to https://tickets.puppetlabs.com/browse/SERVER-94 ?

**Comment by gohoyer** [comment]
I had the same problem and, in my case, it was solved changing the ruby version to 1.8.7.

**Comment by jairojunior** [comment]
It's probably won't be a problem with version 1.1.0
